httperfpy/httperfpy.py

```diff
@@ -117,55 +116,146 @@ def __params(self):
             "wset": None
         }
 
-
 class HttperfParser(object):
 
-    @classmethod
-    def parse(self, result_string):
+    def parse(self, result_string, options = {}):
+        # constants for the parse
+        self.exps = self.__expressions()
+        vexpression = re.compile(
+            "^Connection lifetime = ([-+]?[0-9]*\.?[0-9]+)")
+
+        # items that are accumulated during the parse
+        verbose_connection_times = []
+        verbose_output_lines = []
+        unknown_lines = []
+        xtrace_headers = []
+
+        # helper functions for oddball lines. they're mostly
+        # oddball because i don't know where they can appear
+        # in the output so i check each line for them.
+        def get_verbose_connection_time(line):
+            if (vexpression.match(line)):
+                verbose_connection_times.append(match.group(1))
+            else:
+                unknown_lines.append(line)
+
+        def get_verbose_output_line(line):
+            verbose_output_lines.append(line)
+
+        special_starts = {
+            "Connection lifetime =": get_verbose_connection_time,
+            "httperf: ": get_verbose_output_line
+        }
+
+        # the return value - httperf's output, parsed, as a dict
+        self.matches = {}
+
+        # standard states with transitions
+        transitions = self.__transitions()
+
+        # initial state
+        want = 'command'
+
+        def change_want(new_want):
+            nonlocal want
+            if ('state-changes' in options and options['state-changes']):
+                print("state change {} => {}".format(want, new_want))
+            want = new_want
 
-        verbose_expression = re.compile("^Connection lifetime = ([0-9]+\.[0-9]+)(\s?)", re.M|re.I)
 
         lines = result_string.split("\n")
-        matches = {}
 
-        verbose_connection_times = []
+        i = 0
+        while i < len(lines):
+            line = lines[i]
+            i += 1
 
-        include_header_expressions = self.__has_reply_header(result_string)
+            if line == "":
+                continue
 
-        for line in lines:
+            found = False
+            for start in special_starts:
+                if line.startswith(start):
+                    special_starts[start](line)
+                    found = True
+                if (found):
```

Comment:
Is this if statement needed -- can the `break` just be added right after setting `found = True`?

Reply:
looks like i don't need it.